content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+
+# Redfish® iLO RESTful API as a solution?
+
+Each HPE ProLiant server has a lot of temperature sensors spread across the system board, CPU, memory and power supplies. This detailed temperature data is available and visualized on each ILO interface in the Power&Thermal section:
+
+![](/img/ilotemperature.png "ILO Temperature Information")
+
+The first sensor, 01-Inlet Ambient temperature, is measuring the inlet temperature at the front of the server and can be used as an indicator of the CTC temperature.  Now that I have identified a usable data point an automated reading of the temperature values is needed. Having used the SimpliVity REST API to build the SimpliVity-Prometheus connector, that is used to monitor the core infrastructure, it was a natural step for me to look into the [Redfish® iLO RESTful API](https://developer.hpe.com/platform/ilo-restful-api/home/). The Redfish® API ecosystem is an open industry-standard specification (published by the Distributed Management Task Force [DTMF](http://www.dmtf.org/standards/redfish)) and provides remote server provisioning, inventory and monitoring.
+
+A directory of the available Redfish® resources at an interface can be retrieved with the following RESTful API command: 
+
+\`GET {{baseUrl}}/redfish/v1/resourcedirectory\`
+
+I used [Postman](https://www.postman.com/) as an developer tool to retrieve and search the resource directory. The above RESTful API comand was used in Postman with a basic username/password authorization at one of the ILOs in the CTC, where the baseUrl is defined as <https://iloIPaddress>. I searched in the obtained resource directory for "thermal" in order to find the correct command to retrieve thermal data:
+
+![](/img/redfishresourcedirectory.png "Redfish resource directory (Postman output)")
+
+Hence, I can retrieve thermal server data with the following RESTful command:
+
+\`GET /redfish/v1/chassis/1/thermal\`
+
+And using this Get command with Postman confirmed that I do get the values of all temperature sensors of the server:
+
+![](/img/redfishtemparture.png "ILO Redfish thermal information (Postman output)")
+
+# Building the HPE Server Exporter
+
+Great, now that I do know the RESTful command to retrieve the data I am looking for and I do have a Prometheus-Grafana based monitoring established in the Customer Technology Center, I was only missing a Prometheus exporter for the ILO interfaces and the according Grafana dashboards. Hence I needed to write another Prometheus exporter: one for the ILO interface. I decided to use Python as the programming language for this connector, because I could build on the knowledge gained when I developed the SimpliVity-Prometheus exporter. 
```

Comment:
Again, remember you're describing something that happened in the past. So this should read instead like this:
"Once I had the Prometheus-Grafana based monitoring established int he Customer Technology Center and knew the RESTful command required to retrieve the data I was looking for, the only thing missing was a Prometheus exporter for the ILO interfaces and the according Grafana dashboards. This required me to write another Prometheus exporter - one designed for the ILO interface. "